.github/workflows/sync.yml

```diff
@@ -0,0 +1,45 @@
+name: Sync Dependencies
+on:
+  pull_request:
+    branches:
+      - 'main'
+    types:
+      - 'opened'
+      - 'synchronize'
+      - 'reopened'
+jobs:
+  sync-dependencies:
+    if: startsWith(github.actor, 'dependabot')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21.0
+      - name: Sync all dependencies
+        run: |
+          find . -name 'go.mod' -execdir go mod tidy \;
+      - name: Check for changes
+        id: git_status
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+            echo "has_changes=true" >> $GITHUB_ENV
+          else
+            echo "has_changes=false" >> $GITHUB_ENV
+          fi
+      - name: Commit and push changes
```

Comment:
I can't see the commit created by the [test action run](https://github.com/LS6-Events/astra/actions/runs/9390990172/job/25862107659) on this PR.
Was the commit pushed, has it just been purged/reset?

Reply:
@LBishop234 Yeah exactly I force pushed `6cdacdf` when I had worked out all of the kinks in the pipeline and uncommented the check if the actor was dependabot.

Reply:
Given were squashing merge commits just helps reviews to leave all the messy commits imo.